x/nexus/exported/types.go

```diff
@@ -17,7 +19,15 @@ import (
 	"github.com/axelarnetwork/utils/slices"
 )
 
-//go:generate moq -out ./mock/types.go -pkg mock . MaintainerState
+//go:generate moq -out ./mock/types.go -pkg mock . MaintainerState LockableCoin
+
+// LockableCoin defines a coin that can be locked and unlocked
+type LockableCoin interface {
+	GetCoin() sdk.Coin
+	GetOriginalCoin(ctx sdk.Context) sdk.Coin
+	Lock(ctx sdk.Context, fromAddr sdk.AccAddress) error
+	Unlock(ctx sdk.Context, toAddr sdk.AccAddress) error
```

Comment:
```suggestion
	LockFrom(ctx sdk.Context, fromAddr sdk.AccAddress) error
	UnlockTo(ctx sdk.Context, toAddr sdk.AccAddress) error
```
maybe a bit more readable?